src/components/Drawer.js

```diff
@@ -0,0 +1,146 @@
+const React = require('react');
+const Velocity = require('velocity-animate');
+
+const StyleConstants = require('../constants/Style');
+
+const Icon = require('../components/Icon');
+
+const Drawer = React.createClass({
+  propTypes: {
+    duration: React.PropTypes.number,
+    easing: React.PropTypes.array,
+    isOpen: React.PropTypes.bool,
+    navContent: React.PropTypes.shape({
+      duration: React.PropTypes.number,
+      label: React.PropTypes.string.isRequired,
+      onNextClick: React.PropTypes.func.isRequired,
+      onPreviousClick: React.PropTypes.func.isRequired
+    }),
+    onClose: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps () {
+    return {
+      duration: 500,
+      isOpen: false,
+      easing: [0.28, 0.14, 0.34, 1.04]
+    };
+  },
+
+  componentDidMount () {
+    this._renderTransition(this.props.isOpen);
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.isOpen !== this.props.isOpen) {
+      this._renderTransition(newProps.isOpen);
+    }
+  },
+
+  _renderNavContent () {
+    const styles = this.styles();
+
+    return (
+      <div ref={(ref) => (this._navContent = ref)} style={styles.navContent}>
+        <Icon onClick={this.props.navContent.onPreviousClick} size={25} style={styles.icon} type='caret-left'/>
+        {this.props.navContent.label}
+        <Icon onClick={this.props.navContent.onNextClick} size={25} style={styles.icon} type='caret-right'/>
+      </div>
+    );
+  },
+
+  _renderTransition (isOpen) {
+    const el = this._component;
```

Comment:
@paniclater Looking at your use cases for this, they are all attached to either `div` or `span`. Basically, you could do 
```javascript
<div ref='_navContent' style={styles.navContent}>
```
instead of 
```javascript
<div ref={(ref) => (this._navContent = ref)} style={styles.navContent}>
```
because this is a simple `div` that has no methods or anything you can call using the ref.

Quoted from https://facebook.github.io/react/docs/more-about-refs.html
When attaching a ref to a DOM component like `<div />`, you get the DOM node back; when attaching a ref to a composite component like `<TextInput />`, you'll get the React class instance. In the latter case, you can call methods on that component if any are exposed in its class definition.